src/ocean/net/ssl/openssl/OpenSsl.d

```diff
@@ -0,0 +1,833 @@
+
+    ***************************************************************************/
+
+    public struct SSL_CTX;
+
+
+    /***************************************************************************
+
+        Opaque struct which implements an X509 certificate
+
+    ***************************************************************************/
+
+    public struct X509;
+
+
+    /***************************************************************************
+
+        The context used wile verifying an X509 certificate
+
+    ***************************************************************************/
+
+    public struct X509_STORE_CTX;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds RSA encryption parameters
+
+    ***************************************************************************/
+
+    public struct RSA;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds a message digest
+
+    ***************************************************************************/
+
+    public struct EVP_MD;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds a message digest context
+
+    ***************************************************************************/
+
+    public struct EVP_MD_CTX;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds a private key
+
+    ***************************************************************************/
+
+    public struct EVP_PKEY;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds a private key context
+
+    ***************************************************************************/
+
+    public struct EVP_PKEY_CTX;
+
+
+    /***************************************************************************
+
+        Opaque struct which holds a digest encryption engine
+
+    ***************************************************************************/
+
+    public struct ENGINE;
+
+
+    /***************************************************************************
+
+        Retrieve the function pointers for SSLv3, or v2 if v3 is unavailable
+
+        Returns;
+            A struct with all of the function pointers pointing to the SSL v3
+            functions, falling back to v2 if v3 is unavailable
+
+    ***************************************************************************/
+
+    public ssl_method_st* SSLv23_method();
+
+
+    /***************************************************************************
+
+        Initialize the SSL library by registering algorithms
+
+        Returns:
+            Always returns 1.
+
+    ***************************************************************************/
+
+    public int SSL_library_init ();
```

Comment:
Which OpenSSL version are you binding to ?
This one is deprecated in v1.1.0 (which has a couple of breaking changes) but v1.1.0 is only available from Bionic (18.04), Xenial ships 1.0.2

Reply:
v1.0.x. We need to support Xenial.

Reply:
We need to support both :/
This will need compile-time version flag to chose between 1.1 and 1.0

Reply:
The call still works in Bionic, though.

Reply:
In Bionic, calling this function is no longer necessary, the library performs most initialization automatically. It is still OK to call it, and that preserves backwards compatibility.